pkg/runtime/runtime.go

```diff
@@ -678,19 +683,30 @@ func (a *DaprRuntime) subscribeTopic(parentCtx context.Context, name string, top
 
 	ctx, cancel := context.WithCancel(parentCtx)
 	policy := a.resiliency.ComponentInboundPolicy(ctx, name, resiliency.Pubsub)
-	err := a.pubSubs[name].component.Subscribe(ctx, pubsub.SubscribeRequest{
+	var err error
```

Comment:
If you move this inside the `if` block below, then you should be able to not change all errors inside the method to `innerErr`. I am not a fan of that because it will be error-prone: I remember in previous reviews we caught that the change from `err` to `innerErr` was missed sometimes.

Reply:
Addressed with https://github.com/dapr/dapr/pull/5398